core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -243,12 +253,6 @@ private boolean collides(TypeCodec<?> newCodec, TypeCodec<?> oldCodec) {
       }
     }
 
-    if (value instanceof TupleValue) {
```

Comment:
This wasn't enough for collections of tuples or UDTs. We need to fall through and let `inspectType` + `getCachedCodec` do their jobs in all cases.